notation.go

```diff
@@ -152,6 +172,64 @@ func Sign(ctx context.Context, signer Signer, repo registry.Repository, signOpts
 	return targetDesc, nil
 }
 
+// SignBlob signs the arbitrary data and returns the signature
+func SignBlob(ctx context.Context, signer BlobSigner, blobReader io.Reader, signBlobOpts SignBlobOptions) ([]byte, *signature.SignerInfo, error) {
+	// sanity checks
+	if err := validate(signer, signBlobOpts.SignerSignOptions); err != nil {
+		return nil, nil, err
+	}
+
+	if blobReader == nil {
+		return nil, nil, errors.New("blobReader cannot be nil")
+	}
+
+	if signBlobOpts.ContentMediaType == "" {
+		return nil, nil, errors.New("content media-type cannot be empty")
+	}
+
+	if _, _, err := mime.ParseMediaType(signBlobOpts.ContentMediaType); err != nil {
+		return nil, nil, fmt.Errorf("invalid content media-type '%s': %v", signBlobOpts.ContentMediaType, err)
+	}
+
+	getDescFun := func(hashAlgo digest.Algorithm) (ocispec.Descriptor, error) {
+		h := hashAlgo.Hash()
+		bytes, err := io.Copy(hashAlgo.Hash(), blobReader)
+		if err != nil {
+			return ocispec.Descriptor{}, err
+		}
```

Comment:
Consider [Algorithm.FromReader()](https://pkg.go.dev/github.com/opencontainers/go-digest#Algorithm.FromReader)

Reply:
We cannot use `[Algorithm.FromReader()](https://pkg.go.dev/github.com/opencontainers/go-digest#Algorithm.FromReader)` because we also need content length.